### PR TITLE
Dual-channel-replication: set connection to blocking after blocking-w…

### PR DESCRIPTION
**Pull request by xbasel**
…rite.

Dual-channel file descriptors are set to blocking mode in the main thread, followed by a blocking write. This sets the file descriptors to non-blocking if TLS is used (see `connTLSSyncWrite()`).

The child process runs in blocking mode. If a write operation fails to write the entire buffer, SSL returns `SSL_ERROR_WANT_WRITE`. This error is ignored, causing the replica to fail in loading the RDB.

This change sets file descriptors after the blocking write.

### CODE REVIEW

File: src/rdb.c [hunk 1]
```diff
@@ -3557,15 +3557,15 @@ int rdbSaveToReplicasSockets(int req, rdbSaveInfo *rsi) {
             if (replica->replica_req != req) continue;
 
             conns[connsnum++] = replica->conn;
-            if (dual_channel) {
-                /* Put the socket in blocking mode to simplify RDB transfer. */
-                connBlock(replica->conn);
+            if (dual_channel) {
```
Comment (madolson) [inline src/rdb.c]:
```suggestion
            if (dual_channel) {
```

File: src/rdb.c [hunk 2]
```diff
@@ -3558,14 +3558,14 @@ int rdbSaveToReplicasSockets(int req, rdbSaveInfo *rsi) {
...
             if (dual_channel) {
-                /* Put the socket in blocking mode to simplify RDB transfer. */
-                connBlock(replica->conn);
                 connSendTimeout(replica->conn, server.repl_timeout * 1000);
                 /* This replica uses diskless dual channel sync, hence we need
                  * to inform it with the save end offset.*/
                 sendCurrentOffsetToReplica(replica);
                 /* Make sure repl traffic is appended to the replication backlog */
                 addRdbReplicaToPsyncWait(replica);
+                /* Put the socket in blocking mode to simplify RDB transfer. */
```
Comment (enjoy-binbin) [inline src/rdb.c:3567]: Let's mention this "after" in the comments?

**Review by enjoy-binbin — APPROVED** [review]

File: src/rdb.c (hunk 2 above)
Comment (xbasel) [reply src/rdb.c:3567]: I'm not sure I understand.

Comment (enjoy-binbin) [reply src/rdb.c:3567]: sorry my bad. I means add a comment to explain this after thing

**Comment by madolson** [comment]
100% of the TLS tests are still failing, those should be fixed before we merge this.

**Comment by madolson** [comment]
@xbasel We can close this right?

**Comment by enjoy-binbin** [comment]
closing since #837 had this change and cover it

**Pull request closed without merging**